internal/service/ec2/ec2_fleet.go

```diff
@@ -576,14 +573,23 @@ func resourceFleetCreate(ctx context.Context, d *schema.ResourceData, meta inter
 	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
 
 	fleetType := d.Get("type").(string)
+	excessCapacityTerminationPolicy := d.Get("excess_capacity_termination_policy")
 	input := &ec2.CreateFleetInput{
-		ExcessCapacityTerminationPolicy:  aws.String(d.Get("excess_capacity_termination_policy").(string)),
 		ReplaceUnhealthyInstances:        aws.Bool(d.Get("replace_unhealthy_instances").(bool)),
 		TagSpecifications:                tagSpecificationsFromKeyValueTags(tags, ec2.ResourceTypeFleet),
 		TerminateInstancesWithExpiration: aws.Bool(d.Get("terminate_instances_with_expiration").(bool)),
 		Type:                             aws.String(fleetType),
 	}
 
+	if v, ok := d.GetOk("type"); ok {
+		if v == ec2.FleetTypeMaintain {
+			if excessCapacityTerminationPolicy == nil {
+				d.Set("excess_capacity_termination_policy", ec2.FleetExcessCapacityTerminationPolicyTermination)
```

Comment:
If this is how you are attempting to add a default, the schema is where defaults should be set, not inline in the CRUDL handlers.